google-cloud-core/src/main/java/com/google/cloud/Identity.java

```diff
@@ -256,27 +280,10 @@ public String strValue() {
   public static Identity valueOf(String identityStr) {
     String[] info = identityStr.split(":");
     Type type = Type.valueOf(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, info[0]));
-    switch (type) {
-      case ALL_USERS:
-        return Identity.allUsers();
-      case ALL_AUTHENTICATED_USERS:
-        return Identity.allAuthenticatedUsers();
-      case USER:
-        return Identity.user(info[1]);
-      case SERVICE_ACCOUNT:
-        return Identity.serviceAccount(info[1]);
-      case GROUP:
-        return Identity.group(info[1]);
-      case DOMAIN:
-        return Identity.domain(info[1]);
-      case PROJECT_OWNER:
-        return Identity.projectOwner(info[1]);
-      case PROJECT_EDITOR:
-        return Identity.projectEditor(info[1]);
-      case PROJECT_VIEWER:
-        return Identity.projectViewer(info[1]);
-      default:
-        throw new IllegalStateException("Unexpected identity type " + type);
+    if (type.equals(Type.ALL_AUTHENTICATED_USERS) || type.equals(Type.ALL_USERS)) {
+      return new Identity(type, null);
+    } else {
+      return new Identity(type, checkNotNull(info[1]));
```

Comment:
Yeah, that was inherited from the existing code - do you feel I need to fix it in this PR?